arrow-core-data/src/main/kotlin/arrow/core/Eval.kt

```diff
@@ -547,7 +562,13 @@ sealed class Eval<out A> : EvalOf<A> {
     override fun value(): A = result.getOrElse {
       evaluate(eval).also { result = Some(it) }
     }
+
+    override fun toString(): String =
+      "Eval.Memoize($eval)"
   }
+
+  override fun toString(): String =
```

Comment:
I was wondering why we need the `toString` method in the sealed class

Reply:
I guess it's not needed if it's defined for all the cases.